pkg/controller/status/patch.go

```diff
@@ -0,0 +1,39 @@
+package status
+
+import (
+	"context"
+	"encoding/json"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type patchValue struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+// PatchUpdateStatus performs the JSONPatch patch update to the Atlas Custom Resource.
+// The "jsonPatch" merge allows to update only status field so is more
+func PatchUpdateStatus(kubeClient client.Client, resource mdbv1.AtlasCustomResource) error {
```

Comment:
[non-blocking] I recently found that defining interfaces in the same package in which they are used (even in the same file where they may be needed) and only with the methods I need to be a great way for SOLID, this may lead to have a lot of smaller interfaces around but I'm ok with the tradeoff than having a single big interface

Reply:
sure, do you see anything that could be optimized here? Is it the `AtlasCustomResource`?

Reply:
yes, I'm wondering if it would make sense to move it from `pkg/api/v1/atlascustomresource.go` to this same file but up to you, also if you follow vasilevp comment on static checking it may be better to leave it where it is to avoid a circular dependency

Reply:
yes, this is the central interface residing in the `apis` package which doesn't depend on anything else. But `controller` package depends on `apis` so moving it elsewhere will turn into circular dependency.

But thanks for the suggestion, I agree that having "local" interfaces looks like a common Go pattern 👍